packages/openssl.rb

```diff
@@ -65,10 +65,32 @@ def self.check
     return if ARCH == 'i686'
 
     # Don't run tests if we are just rebuilding the same version of openssl.
-    system 'make test' unless `openssl version | awk '{print $2}'`.chomp == '1.1.1s'
+    system 'make test' unless `openssl version | awk '{print $2}'`.chomp == version.to_s
   end
 
   def self.install
     system "make DESTDIR=#{CREW_DEST_DIR} install_sw install_ssldirs"
+    # Extract OpenSSL 1.1.1t libraries for backwards compatibility purposes.
+    # Builds and rebuilds of packages against OpenSSL should automatically
+    # build against OpenSSL 3.x and not against OpenSSL 1.1.1x.
+    File.write 'openssl111_files', <<~EOF
+      usr/local/#{ARCH_LIB}/libcrypto.so.1.1
+      usr/local/#{ARCH_LIB}/libssl.so.1.1
+    EOF
+    @cur_dir = `pwd`.chomp
+    case ARCH
+    when 'aarch64', 'armv7l'
+      downloader 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/1.1.1t_armv7l/openssl-1.1.1t-chromeos-armv7l.tar.xz',
+                 '88a36d1539c7c01af1f5e469b64c2760f43126bb75c0e63b53d3d61c2a6fbe7f', 'openssl-1.1.1t-chromeos.tar.xz'
+    when 'i686'
+      downloader 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/1.1.1t_i686/openssl-1.1.1t-chromeos-i686.tar.xz',
+                 'ce98c1898e57df1cbcceab08912e219fb5f27b0e4585315f7babdf524fa844dc', 'openssl-1.1.1t-chromeos.tar.xz'
+    when 'x86_64'
+      downloader 'https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/1.1.1t_x86_64/openssl-1.1.1t-chromeos-x86_64.tar.xz',
+                 'd6583dc2c7566da33402cb8d7f9025189d65b8ba6300e752edcb0d74ebcb1f68', 'openssl-1.1.1t-chromeos.tar.xz'
```

Comment:
There are a lot of redundant versions here.  Would it make sense to use `#{version}`?

Reply:
Code suggestion to replace this block?

Reply:
```
    @legacy_version = '1.1.1t'
    case ARCH
    when 'aarch64', 'armv7l'
      downloader "https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/#{@legacy_version}_armv7l/openssl-#{@legacy_version}-chromeos-armv7l.tar.xz",
                 '88a36d1539c7c01af1f5e469b64c2760f43126bb75c0e63b53d3d61c2a6fbe7f', "openssl-#{@legacy_version}-chromeos.tar.xz"
    when 'i686'
      downloader "https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/#{@legacy_version}_i686/openssl-#{@legacy_version}-chromeos-i686.tar.xz",
                 'ce98c1898e57df1cbcceab08912e219fb5f27b0e4585315f7babdf524fa844dc', "openssl-#{@legacy_version}-chromeos.tar.xz"
    when 'x86_64'
      downloader "https://gitlab.com/api/v4/projects/26210301/packages/generic/openssl/#{@legacy_version}_x86_64/openssl-#{@legacy_version}-chromeos-x86_64.tar.xz",
                 'd6583dc2c7566da33402cb8d7f9025189d65b8ba6300e752edcb0d74ebcb1f68', "openssl-#{@legacy_version}-chromeos.tar.xz"
...
system "tar -xv --files-from #{@cur_dir}/openssl111_files -f #{@cur_dir}/openssl-#{@legacy_version}-chromeos.tar.xz"
```

Reply:
Done.